tests/utils.py

```diff
@@ -0,0 +1,29 @@
+from contextlib import contextmanager
+import logging
+
+
+def retry(
+    operation,
+    times=1,
+    wait=1,
+    error_msg=None,
+    logger='tests.utils.retry',
+):
+    log = logging.getLogger(logger)
+
+    for idx in range(times):
+        try:
+            operation()
+            log.info("Attempt %d/%d succeeded", idx, times)
```

Comment:
Use an `else` statement to differentiate what needs to be part of the `try` statement and what does not.